src/context/MessagingContext.js

```diff
@@ -34,26 +34,30 @@ export const MessagingProvider = (props) => {
     let newMessages = rawMessages
       .sort((a, b) => (a.composed_at < b.composed_at ? -1 : 1))
       .reduce((grouped, message) => {
-        if (
-          message.subject === 'Message' ||
-          message.subject === 'Announce Message'
-        ) {
-          let key =
-            message.to[0].recipient !== user.userName
-              ? message[`to`][0].recipient
-              : message['from'].author;
+        if (message.type === 'message') {
+          let key = message.to !== user.userName ? message.to : message.from;
           if (key) {
             if (!grouped[key] && !messages[key]) {
               grouped[key] = [];
             }
             grouped[key].push(message);
           }
-        } else if (message.subject === 'Survey') {
+        } else if (message.type === 'survey') {
           let key = message.survey.id;
           if (!grouped[key]) {
             grouped[key] = [];
           }
           grouped[key].push(message);
+        } else if (message.type === 'announce') {
+          // assume it's an announcement
+          let key = message.subject;
+          if (grouped[key]) {
+            const date = Date.now();
+            console.log('date', date);
+            grouped[`${key}-${Date.now()}`] = [message];
+          } else {
```

Comment:
hmm ok i see, maybe that group key using Date.now() is handling the grouping logic I mentioned above?

Reply:
If the bulk messages have ids then we can group them by that unique id and not worry about adding the date to the name.  Surveys have the id, but I don't think the announce messages do.

It might be interesting to group all the announcements by organization/region id too...